openspp-project/openspp-registry
# [FIX] spp_api: Override the handle_error function

## **Why is this change needed?**
To fix the `spp_api` initialization error.

## **How was the change implemented?**
Added and overridden the function `handle_error` since not overriding this is the cause of the api not initializing properly.

## **New unit tests**
```
None
```

## **Unit tests executed by the author**
```
None
```

## **How to test manually**
In Odoo:
- Install or Upgrade `spp_api`.
- Install `spp_api_records`.
- Activate Developer Mode.
- Navigate to Users and Find your User Account.
- In Allowed APIs tab check if `spp_api` is in the list.
- Click the Key Icon in the right side of OpenAPI Token.
- A pop-up will open, click the Copy. (Paste this on a notepad this is important, this is the `bearer token`.)

In Postman: (Download this postman collection: 
[OpenSPP-ACF API.postman_collection.json](https://github.com/user-attachments/files/17844843/OpenSPP-ACF.API.postman_collection.json))

- Open the postman.
- Import the postman collection.
- Click the created collection from the import (OpenSPP-ACF API).
- Go to Variables.
- In (Current Value) `user_prog_token` and `user_reg_token` replace the value after 'Bearer' with the `bearer token` you copied before. (Make sure the word `Bearer and a blank space` is still there before the `bearer token` you pasted.).
e.g Bearer ey........
- In (Current Value) `api_url` change the url but make sure the `/api/spp_api/1/` is still in the correct place.
- Then click on `Get the list of programs` or anything from the list then click `Send`.

## **Notes**
I'm not sure how or if we can test this with runboat as a `private_key.pem` and `public_key.pub` is needed to be stored inside the container's `/etc/secrets/` folder.

## **Related links**
https://github.com/OpenSPP/openspp-modules/issues/603
